cl/donate/signals.py

```diff
@@ -0,0 +1,17 @@
+from django.conf import settings
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from cl.donate.models import NeonWebhookEvents
+
+
+@receiver(
+    post_save,
+    sender=NeonWebhookEvents,
+    dispatch_uid="truncate_webhook_events",
+)
+def truncate_webhook_events(sender, instance=None, created=False, **kwargs):
+    if not instance.id % settings.NEON_MAX_WEBHOOK_NUMBER:
+        NeonWebhookEvents.objects.filter(
+            pk__lte=instance.id - settings.NEON_MAX_WEBHOOK_NUMBER
+        ).delete()
```

Comment:
We also want things that could violate our privacy policy, so, something like:

```suggestion
        twelve_weeks_ago = XYZ
        NeonWebhookEvents.objects.filter(
            Q(pk__lte=instance.id - settings.NEON_MAX_WEBHOOK_NUMBER) |
              Q(date_created__lt=twelve_weeks_ago),
        ).delete()
```

Reply:
understood!